project_sla/project_sla_control.py

```diff
@@ -320,9 +325,21 @@ class SLAControlled(orm.AbstractModel):
     }
 
     def store_sla_control(self, cr, uid, ids, context=None):
-        docs = self.browse(cr, uid, ids, context=context)
-        self.pool.get('project.sla.control').store_sla_control(
-            cr, uid, docs, context=context)
+        if ids:
+            try:
+                cr.execute("SAVEPOINT %s_sla_save" % self._table)
+                cr.execute("""
+                    SELECT * FROM %s WHERE id IN %%s
+                    FOR UPDATE of %s NOWAIT
+                """ % (self._table, self._table), (tuple(ids),))
+            except:
+                cr.execute("ROLLBACK TO %s_sla_save" % self._table)
+                _logger.error("Cannot lock table %s", self._table)
+                raise
```

Comment:
No problem if it gets updated again. Each individual Document is independent, so no real need enforce transaction integrity.